components/server/src/ome/security/policy/Policy.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2015 University of Dundee & Open Microscopy Environment.
+ * All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+package ome.security.policy;
+
+import java.util.Set;
+
+import ome.conditions.SecurityViolation;
+import ome.model.IObject;
+
+/**
+ * Strategy to permit the flexible restriction of certain actions
+ * throughout OMERO. Code which intends to allow such checks should
+ * create a new {@link Policy} class with a unique {@link #getName() name}
+ * and add checks within that code. For example:
+ *
+ * <pre>
+ *    public class MyPolicy implements Policy {
+ *
+ *        public final static string NAME = "MyPolicy";
+ *
+ *        public String getName() {
+ *            return NAME;
+ *        }
+ *    }
+ *
+ *    public void someImportantMethod() {
+ *        IObject objBeingAccessed = ...;
+ *        policyService.checkRestriction(MyPolicy.NAME, objBeingAccessed);
+ *        // Here an exception may be thrown
+ *    }
+ * </pre>
+ */
+public interface Policy {
+
+    /**
+     * Unique name for this type of {@link Policy}. This string will be sent to
+     * clients via
+     * {@link ome.model.internal.Permissions#copyExtendedRestrictions()} in
+     * order to prevent exceptions.
```

Comment:
Not sure I have anything to suggest here, but this made sense only when I realized that it is the policy names that get passed as the extended restrictions, it's more than a name to put in a log message or whatever.